ilp-core/src/main/java/org/interledger/core/InterledgerAddress.java

```diff
@@ -67,9 +57,8 @@ static InterledgerAddress of(final String value) {
    */
   static boolean isValid(final String value) {
     Objects.requireNonNull(value);
-    return (value.endsWith(".") ? ADDRESS_PREFIX_PATTERN : DESTINATION_ADDRESS_PATTERN)
-        .matcher(value)
-        .matches();
+    ADDRESS_PARSER.validate(value);
```

Comment:
To illustrate the discussion about the "cause of `IllegalArgumentException`" check in the `InterledgerAddressBuilderTest` class, I would change this line into:

> try {
>   ADDRESS_PARSER.validate(value);
> } catch (final IllegalArgumentException e) {
>   throw new IllegalArgumentException("Address is invalid", e)
> }

Then `InterledgerAddressBuilderTest` would only rely on that later message.